File: leetcode/algorithms/tree/insert-into-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function insertIntoBST(root: TreeNode | null, val: number): TreeNode | null {
    
    if(root===null){
        const freshNode = new TreeNode(val);
        root = freshNode;
    } else {
        let walker: TreeNode = root;
        while(root!==null){
            if(val<walker.val){
                if(walker.left===null){
                    const freshNode = new TreeNode(val);
                    walker.left = freshNode;
                    break;
                } else {
                    walker = walker.left;
                }
            } else {
                if(walker.right===null){
                    const freshNode = new TreeNode(val);
                    walker.right = freshNode;
                    break;
                } else {
                    walker = walker.right;
                }
            }
        }
    }
    
    return root;
    
    
};

File: leetcode/algorithms/tree/insert-into-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private void addNewNodeToBST(TreeNode root,int val){
        if(root==null){
            return;
        }
        if(val<=root.val){
            if(root.left==null){
                root.left = new TreeNode(val);
            } else {
                addNewNodeToBST(root.left,val);
            }
        } else {
            if(root.right==null){
                root.right = new TreeNode(val);
            } else {
                addNewNodeToBST(root.right,val);
            }
        }
    }
    
    public TreeNode insertIntoBST(TreeNode root, int val) {
        
        addNewNodeToBST(root,val);
        return root==null ? new TreeNode(val) : root;
        
    }
}